src/components/MerchantsPage/gam/GiftAMealPage.tsx

```diff
@@ -21,41 +21,56 @@ const GiftAMealPage = (props: Props) => {
 
   const [activeCampaigns, setActiveCampaigns] = useState<Campaign[]>([]);
   const [pastCampaigns, setPastCampaigns] = useState<Campaign[]>([]);
+  const [currPage, setCurrPage] = useState(0);
+  const [totalPages, setTotalPages] = useState<number>(0);
+  const [totalCountPastCmpgn, setTotalCountPastCmpgn] = useState(0);
+  const [shouldFetchPastData, setShouldFetchPastData] = useState(true);
   const [selectedCampaignId, setSelectedCampaignId] = useState<number | null>(
     null
   );
 
   const fetchData = async () => {
-    const campaignResponse = await getCampaigns();
+    const { data: campaignData } = await getCampaigns();
     const activeMegaGam: Campaign[] = [];
     const active: Campaign[] = [];
-    const past: Campaign[] = [];
 
     // Campaigns come back sorted by oldest to newest end date.
-    campaignResponse.data.forEach((campaign: Campaign) => {
-      if (campaign.active && campaign.valid) {
-        if (campaign.project_id) {
-          activeMegaGam.push(campaign);
-        } else {
-          active.push(campaign);
-        }
+    campaignData.forEach((campaign: Campaign) => {
+      if (campaign.project_id) {
+        activeMegaGam.push(campaign);
       } else {
-        past.push(campaign);
+        active.push(campaign);
       }
     });
 
     // We want active Mega GAM campaigns to show up first and then active
     // campaigns from oldest to newest.
     setActiveCampaigns(activeMegaGam.concat(active));
-    // Inactive campaigns should show up newest to oldest.
-    setPastCampaigns(past.reverse());
   };
 
   useEffect(() => {
     fetchData();
     // eslint-disable-next-line
   }, []);
 
+  useEffect(() => {
```

Comment:
What's the reason for adding another `useEffect()` call? Can we combine these?

Reply:
Don't think so, cuz the way I have the effect for fetching past campaigns set up is that it needs some state in order to synchronize, whereas the first one doesn't really depend on anything to synchronize.